helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -235,7 +235,37 @@ receivers:
               from: log
               to: $$
   {{- end }}
-
+  # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver
+  {{- if (eq (include "splunk-otel-collector.logsEnabled" .) "true") }}
+  {{- if and (eq .Values.logsEngine "otel") .Values.logsCollection.journaldLogs.enabled }}
+  {{- if .Values.logsCollection.journaldLogs.units }}
+  {{- range $_, $unit := .Values.logsCollection.journaldLogs.units }}
+  {{- printf "journald/%s:" $unit.name | nindent 2 }}
+    directory: {{ $.Values.logsCollection.journaldLogs.directory }}
+    units: [{{ $unit.name }}]
+    priority: {{ $unit.priority | default $.Values.logsCollection.journaldLogs.defaultPriority }}
```

Comment:
Is this field the only reason to have receiver per unit? If you think it's critical to give users an option to customize priority per unit, I recommend grouping them by priority and create receiver per priority, e.g. `journald/info`, `journald/error` etc. I believe usually users don't need different levels for each unit, so it'll be probably just one receiver in most cases --> better for performance. Look at https://masterminds.github.io/sprig/dicts.html and http://masterminds.github.io/sprig/lists.html on how to create the groups from your existing config interface

Reply:
it is needed for `cursor` (checkpoint data) If user is migration from SCK, which stores cursor information per service, and in this format, it will be handled seamlessly. 

so this is how journald cursor works. 
Let's say SCK has read these journald logs. 

message1 | kubelet | cursor1
message2 | docker | cursor2
message3 | ssh | cursor3
message4 | kubelet | cursor4
message5 | docker  | cursor5

After checkpoint migration, we have `cursor3` for ssh, `cursor4` for kubelet, and `cursor5` for docker. 

If i use ssh's cursor info and start reading journal logs, I will get message 4 & 5, and so on. so message 4& 5 would be duplicate. 

the only way to not have duplicate is to give cursor for each service. 
but even if we just choose arbitrary cursor from what we have (among cursor 3~5), it could have some data loss or duplicate, but it won't be a lot. I think it should be acceptable for the performance. `kubelet` service is constantly logging in k8s cluster, so I think it is safe to just take kubelet's cursor. 
Should I change the checkpoint migration script for journald?

Reply:
@rockb1017  I would leave it up to you. Having less receivers is definitely better, but if you think that it'll be hard to ensure checkpoint migration, we can leave it as is and maybe change later

Reply:
Hello, @dmitryax @luckyj5 
I put more thought into it and I think checkpoint migration for journald is not crucial. 
by default we are reading journald logs ignoring old records (`--follow` option) 
So for those who are migrating, they can just deploy olly chart then shutdown sck. Then it will have no data loss, only a slight data dups. 
Instead of having some data dups, we can simplify journald data pipelines and improve performance. 
What do you think?

Reply:
Sounds good to me. I'm always up for simplifying things and improving performance!

Reply:
> Hello, @dmitryax @luckyj5 I put more thought into it and I think checkpoint migration for journald is not crucial. by default we are reading journald logs ignoring old records (`--follow` option) So for those who are migrating, they can just deploy olly chart then shutdown sck. Then it will have no data loss, only a slight data dups. Instead of having some data dups, we can simplify journald data pipelines and improve performance. What do you think?

Thanks for updating. Could you make this change? making it simplied? so instead of having a journald receiver for each service unit, we would have it grouped by priority level.

Reply:
Okay. Will group it my priority. Thanks!

Reply:
This can be addressed in another PR. @luckyj5 feel free to submit an issue

Reply:
Created new issue for tracking this - https://github.com/signalfx/splunk-otel-collector-chart/issues/360